src/spikeinterface/core/analyzer_extension_core.py

```diff
@@ -76,6 +76,19 @@ def _select_extension_data(self, unit_ids):
         new_data["random_spikes_indices"] = np.flatnonzero(selected_mask[keep_spike_mask])
         return new_data
 
+    def _merge_extension_data(
+        self, units_to_merge, new_unit_ids, new_sorting_analyzer, keep_mask=None, verbose=False, **job_kwargs
+    ):
+        new_data = dict()
+        if keep_mask is not None:
+            valid = keep_mask[self.sorting_analyzer.get_extension("random_spikes")._get_data()]
+            nb_skipped = np.cumsum(~keep_mask)
+            new_data["random_spikes_indices"] = np.flatnonzero(valid)
+            new_data["random_spikes_indices"] -= nb_skipped[new_data["random_spikes_indices"]]
```

Comment:
This hard to read and not so fast.
Why not simple boolean mask on the new space ? and then nonzeor on it ?

Reply:
We need to keep only the indices of the valid spikes. Then substract to each value the number of spikes that has been removed before this particular index. But open to suggestion